### PR TITLE
Add an example of dynamic validation groups

### LINKED ISSUE #152: Dynamic validation groups
Can you please provide examples for this scenario, because documentation https://api-platform.com/docs/core/validation#using-validation-groups is not so obvious.

@dunglas @kbsali

### PR DESCRIPTION
**Pull request by dunglas**
Closes #152

### CODE REVIEW

File: core/validation.md [hunk 1]
```diff
@@ -54,6 +54,47 @@ If you need to dynamically determine which validation groups to use for an entit
...
+class Book
+{
+    /**
+     * Return dynamic validation groups.
+     *
+     * @param self $book Contains the instance of Book to validate.
+     *
+     * @return string[]
+     */
+    public static function validationGroups(self $book)
```
Comment (teohhanhui) [inline core/validation.md]: Ideally this should belong in another class, no?

Comment (dunglas) [inline core/validation.md]: It depends of the use case, if it's only some lines of code I think it's ok to put it in the entity.
But the goal here is just to give a short example, introducing a new class is overkill for a doc.

**Comment by aledeg** [comment]
Silly question.
Is it possible to use a service instead of a class? If so, how can I do it?

**Comment by teohhanhui** [comment]
We can make it work with a service with some extra glue (in the Symfony Bridge).

**Comment by dunglas** [comment]
Yes but it's currently not supported.

**Pull request merged**